test/parallel/test-net-server-call-listen-multiple-times.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+// This test checks two cases.
+
+// The first one is about calling a listen method again,
+// after first time is fail (without calling close method).
+
+// The second one is about calling a listen method twice.
+// It should be ELISTENCALLEDTWICE but the first server listen call is working.
+
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
+
+let err1 = false;
+let err2 = false;
+
+function case1() {
+
+  const dummyServer = net.Server();
+  const server1 = net.Server();
+
+  // run some server
+  // in order to simulate EADDRINUSE error
+  dummyServer.listen(0, 'localhost', () => {
+    // try to listen used port
+    server1.listen(dummyServer.address().port, 'localhost');
+  });
+
+  server1.on('error', function(e) {
+    if (e.code === 'EADDRINUSE') {
+      server1.listen(0, 'localhost', () => {
+        dummyServer.close();
+        server1.close();
+      });
+    } else if (e.code === 'ELISTENCALLEDTWICE') {
+      err1 = true;
+    }
+  });
+
+
+}
+
+function case2() {
+  const server2 = net.Server();
+
+  server2.on('error', common.mustCall(function(e) {
+    assert.strictEqual(e.code, 'ELISTENCALLEDTWICE');
+    err2 = true;
```

Comment:
`err2` isn't needed. This listener should only be called once, and is guaranteed by `common.mustCall()`.